sdk/identity/azure-identity/conftest.py

```diff
@@ -159,3 +160,28 @@ def event_loop():
 
     yield loop
     loop.close()
+
+@pytest.fixture(scope="session", autouse=True)
+def add_sanitizers(test_proxy):
+    if EnvironmentVariables.MSI_ENDPOINT in os.environ:
+        url = os.environ.get(EnvironmentVariables.MSI_ENDPOINT)
+        PLAYBACK_URL = "https://msi-endpoint/token"
+        add_general_regex_sanitizer(regex=url, value=PLAYBACK_URL)
+    if "USER_ASSIGNED_IDENTITY_CLIENT_ID" in os.environ:
+        PLAYBACK_CLIENT_ID = "client-id"
+        user_assigned_identity_client_id = os.environ.get("USER_ASSIGNED_IDENTITY_CLIENT_ID")
+        add_general_regex_sanitizer(regex=user_assigned_identity_client_id, value=PLAYBACK_CLIENT_ID)
+    if "CAE_ARM_URL" in os.environ and "CAE_TENANT_ID" in os.environ and "CAE_USERNAME" in os.environ:
+        try:
```

Comment:
Is there a reason for the `try`/`except` logic for the CAE case? Unless there's a situation where these variables are typically present even when tests are skipped, it might be better to raise errors here instead of silently failing to parse the URL.

Reply:
CAE tests are not enabled yet. We want to always continue no matter the setting for CAE.